scripts/install-cni.sh

```diff
@@ -35,7 +35,20 @@ fi
 # Fill CNI spec template.
 token=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
 node_url="https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT}/api/v1/nodes/${HOSTNAME}"
-ipv4_subnet=$(curl -k -s -H "Authorization: Bearer $token" $node_url | jq '.spec.podCIDR')
+ipv4_subnet=""
+if [ -n "${token}" ]; then
+  # use service account token authentication strategy
+  # a service account token should be mounted into the container for this option
+  ipv4_subnet=$(curl -k -s -H "Authorization: Bearer $token" $node_url | jq '.spec.podCIDR')
+else
+  # use client certification authentication strategy
```

Comment:
Should we prefer this strategy, or maybe make the choice explicit with an environment variable? E.g.
```
if [ -n "${CLIENT_KEY_PATH}" ] && [ -n "${CLIENT_CERT_PATH}" ]; then
   ...
```

Reply:
Good point: the client cert authentication should be preferred. Fixed it!

We may still want to set `AutomountServiceAccountToken` to `false` in a future PR after we verify that the client cert authentication works.